File: app/src/main/java/uk/co/glass_software/android/cache_interceptor/demo/volley/VolleyDemoPresenter.java
package uk.co.glass_software.android.cache_interceptor.demo.volley;

import android.content.Context;

import com.android.volley.RequestQueue;
import com.android.volley.toolbox.Volley;

import java.util.concurrent.Callable;

import io.reactivex.Observable;
import uk.co.glass_software.android.cache_interceptor.annotations.CacheInstruction;
import uk.co.glass_software.android.cache_interceptor.demo.DemoPresenter;
import uk.co.glass_software.android.cache_interceptor.demo.model.JokeResponse;
import uk.co.glass_software.android.cache_interceptor.interceptors.RxCacheInterceptor;
import uk.co.glass_software.android.cache_interceptor.interceptors.error.ApiError;

import static uk.co.glass_software.android.cache_interceptor.annotations.CacheKt.DEFAULT_DURATION;

public class VolleyDemoPresenter extends DemoPresenter {

    private final static String URL = BASE_URL + ENDPOINT;
    private final RxCacheInterceptor.Factory<ApiError> rxCacheInterceptorFactory;
    private final RequestQueue requestQueue;

    public VolleyDemoPresenter(Context context,
                               Callable<String> onLogOutput) {
        super(context, onLogOutput);
        requestQueue = Volley.newRequestQueue(context);
        rxCacheInterceptorFactory = RxCacheInterceptor.Companion.buildDefault(context);
    }

    @Override
    protected Observable<JokeResponse> getResponseObservable(boolean isRefresh) {
        CacheInstruction.Operation cacheOperation = new CacheInstruction.Operation.Cache(
                false,
                DEFAULT_DURATION,
                false,
                false
        );

        CacheInstruction instruction = new CacheInstruction(
                JokeResponse.class,
                cacheOperation,
                false,
                false
        );

        return VolleyObservable.Companion.create(
                requestQueue,
                gson,
                rxCacheInterceptorFactory.create(
                        JokeResponse.class,
                        instruction,
                        URL,
                        null
                ),
                URL
        );
    }

    @Override
    public void clearEntries() {
        rxCacheInterceptorFactory.clearOlderEntries();
    }

}

File: app/src/main/java/uk/co/glass_software/android/cache_interceptor/demo/volley/VolleyDemoPresenter.kt
package uk.co.glass_software.android.cache_interceptor.demo.volley

import android.content.Context

import com.android.volley.RequestQueue
import com.android.volley.toolbox.Volley

import io.reactivex.Observable
import uk.co.glass_software.android.boilerplate.lambda.Callback1
import uk.co.glass_software.android.cache_interceptor.demo.DemoPresenter
import uk.co.glass_software.android.cache_interceptor.demo.model.JokeResponse

class VolleyDemoPresenter(context: Context,
                          onLogOutput: (String) -> Unit)
    : DemoPresenter(context, onLogOutput) {

    //    private val rxCacheInterceptorFactory: RxCacheInterceptorFactory<ApiError> = RxCacheInterceptorFactory.buildDefault(context)
    private val requestQueue: RequestQueue = Volley.newRequestQueue(context)

    override fun getResponseObservable(isRefresh: Boolean): Observable<JokeResponse> {
//        val cacheOperation = CacheInstruction.Operation.Expiring.Cache()

//        val instruction = CacheInstruction(
//                JokeResponse::class.java,
//                cacheOperation,
//                false
//        )

        return Observable.error(NoSuchElementException())
//        return VolleyObservable.create(
//                requestQueue,
//                gson,
//                rxCacheInterceptorFactory.create(
//                        JokeResponse::class.java,
//                        instruction,
//                        URL,
//                        null
//                ),
//                URL
//        )
    }

    override fun clearEntries() {
//        rxCacheInterceptorFactory.clearOlderEntries()
    }

    companion object {
        private const val URL = DemoPresenter.BASE_URL + DemoPresenter.ENDPOINT
    }

}